dexcom/fetch/task.go

```diff
@@ -51,7 +51,7 @@ func FailTask(l log.Logger, t *task.Task, err error) error {
 	return err
 }
 
-func shouldTaskError(t *task.Task) bool {
+func shouldTaskFail(t *task.Task) bool {
```

Comment:
I don't think this function handles correctly the case where the `t.Data[dexcomTaskRetryField]` is not set. When `t.Data[dexcomTaskRetryField]` is `nil`, this function should return `false` instead of `true`, so the retry counter can be set to `1`.

Reply:
Yes, that is correct and I intentionally did not fix that yet. 

If we make that fix right now, then every Dexcom task that hits a 429 will start down the "retry path" and, after retrying 3 times, will be marked as failed. I really don't want that to happen quite yet. I need to first straighten out what errors cause a formal retry (where the retry counter is incremented; e.g. Dexcom auth errors), what errors retry for "free" (no increment; e.g. Dexcom 429s), and what errors fail immediately (e.g bad data). Furthermore, I need to add code that resets the retry counter on success (as there is no code to do that currently).